### PR TITLE
add grpc

### PR DESCRIPTION
**Pull request by wolfee001**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/grpc/config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  "1.25.0":
+    folder: all
+  "1.26.0":
+    folder: all
+  "1.27.0":
+    folder: all
+  "1.27.2":
+    folder: all
+  "1.27.3":
+    folder: all
```
Comment (SpaceIm) [inline recipes/grpc/config.yml:11]: I would recommend adding only one version to begin (the latest for example), to quickly check errors with CI.

**Review by Croydon — CHANGES_REQUESTED** [review]
I would appreciate it very much when you would respect the MIT license and give credits when you copy MIT licensed code. This really isn't too much to ask.

Stolen from https://github.com/inexorgame/conan-grpc

Also, I don't think this is a good idea until we figured out protobuf+protoc, e.g. https://github.com/conan-io/conan-center-index/pull/2037, https://github.com/conan-io/conan-center-index/pull/2488

I would migrate the recipe myself afterwards, after having worked on this moving target forever.

**Comment by wolfee001** [comment]
> I would appreciate it very much when you would respect the MIT license and give credits when you copy MIT licensed code. This really isn't too much to ask.
> 
> Stolen from https://github.com/inexorgame/conan-grpc
> 
> Also, I don't think this is a good idea until we figured out protobuf+protoc, e.g. #2037, #2488
> 
> I would migrate the recipe myself afterwards, after having worked on this moving target forever.

Since you didn't reply on my email ent you directly, i wanted to move forward. Missing the credits is my bad, really sorry about that.
That would be nice, if you migrated your solution (as I proposed in my email sent to you about a month ago), but since the two referenced PRs seems to be quite pending, I don't think it would move on soon.
I close the PR if you want me to, but i think this package would be really helpful for the community.

Stolen... pff...

**Comment by Croydon** [comment]
Please only send me emails for sensitive or private requests. I can't provide personal support via email and stopped attempting to do so. Use the respective issue tracker which is https://github.com/inexorgame/conan-grpc for this case.

Like for every Conan recipe I'm working on, the goal is to eventually migrate it to the Conan Center Index (which I also have said on numerous issues and PRs related to gRPC). But many recipes just aren't ready yet as CCI has increased quality demands (which is a good thing) or because they simply don't work well enough yet.

I have updated both my personal profile and the readme of the conan-grpc repository to make those two points clear.

In your email you described your need for a Conan cache / remote and said that Bintray has no binaries for gRPC. This is not true.  The readme directly tells you what the remote is: https://github.com/inexorgame/conan-grpc#add-remote. A quick look in the issue tracker would also have reminded you of that.

If you are allowed to use other Conan remotes from Bintray than the official Conan Center remote, add the inexorgame remote for gRPC itself and the [Bincrafters remote](https://bincrafters.readthedocs.io/en/latest/using_packages.html#adding-the-bincrafters-repository-as-a-conan-remote) for the older Protobuf and Protoc packages and you are good to go.

I'm the core maintainer of the gRPC recipe since 2017 and getting the recipe to support the latest gRPC version and, more importantly, making it actually work rock solid is not as trivial as it might seem on first sight.

The interest in a gRPC package is quite huge and several people said hello and expressed interest to help me move things forward faster, well, every single one of those people vanished after a few days.

gRPC is regularly adding new requirements for which someone needs to create own Conan recipes along the way or updating existing recipes for new versions. We did almost completely caught up on this one, also thanks to much more new contributors since the start of CCI.

Even more of a blocker, many things required new Conan client features, but we caught now up in this area quite good as well.

The last really huge blocker is getting the protobuf+protoc recipe more stable and more up-to-date, but we are close on this one too. Afterwards, gRPC will require similar workarounds for the location problem of executables as the protobuf+protoc recipe and similar heavy patching of the CMake files, so that the mix of generated CMake files + upstream CMake files work well enough together. Related to this: components support (for which I have already a draft). If those are done, I'm confident enough to attempt a migration to CCI and see how it goes.

Though, to get both gRPC and Protobuf rock solid we need more Conan client features to better handle executables (https://github.com/conan-io/conan/issues/7240, https://github.com/conan-io/conan/pull/7324, probably others...)

Cross-compiling support will probably require more work after that as well.

For the moment, I think closing this PR is best.

**Pull request closed without merging**